Source/Core/Core/HW/EXI/EXI_DeviceBaseboard.cpp

```diff
@@ -2,20 +2,133 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 
 #include "Core/HW/EXI/EXI_DeviceBaseboard.h"
+#include <algorithm>
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
 
-#include "Common/Assert.h"
-#include "Common/ChunkFile.h"
+#include <fmt/format.h>
+
+#include "Common/Buffer.h"
 #include "Common/CommonTypes.h"
+#include "Common/Config/Config.h"
+#include "Common/FileUtil.h"
+#include "Common/IOFile.h"
+#include "Common/IniFile.h"
 #include "Common/Logging/Log.h"
+#include "Core/Boot/Boot.h"
+#include "Core/BootManager.h"
+#include "Core/Config/MainSettings.h"
+#include "Core/Config/SYSCONFSettings.h"
+#include "Core/ConfigLoaders/BaseConfigLoader.h"
+#include "Core/ConfigLoaders/NetPlayConfigLoader.h"
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+#include "Core/HW/DVD/AMMediaboard.h"
+#include "Core/HW/DVD/DVDInterface.h"
+#include "Core/HW/EXI/EXI.h"
+#include "Core/HW/EXI/EXI_Device.h"
+#include "Core/HW/MMIO.h"
+#include "Core/HW/Memmap.h"
+#include "Core/HW/SI/SI.h"
+#include "Core/HW/SI/SI_Device.h"
+#include "Core/HW/Sram.h"
+#include "Core/HW/WiimoteReal/WiimoteReal.h"
+#include "Core/Movie.h"
+#include "Core/NetPlayProto.h"
+#include "Core/PowerPC/PowerPC.h"
+#include "Core/System.h"
+#include "Core/WiiRoot.h"
+#include "DiscIO/Enums.h"
+
+static bool g_interrupt_set = false;
+static u32 g_irq_timer = 0;
+static u32 g_irq_status = 0;
+
+static u16 CheckSum(u8* data, u32 length)
+{
+  u16 check = 0;
+
+  for (u32 i = 0; i < length; i++)
+  {
+    check += data[i];
+  }
+
+  return check;
+}
 
 namespace ExpansionInterface
 {
-CEXIBaseboard::CEXIBaseboard(Core::System& system) : IEXIDevice(system)
+
+void GenerateInterrupt(int flag)
+{
+  auto& system = Core::System::GetInstance();
+
+  g_interrupt_set = true;
+  g_irq_timer = 0;
+  g_irq_status = flag;
+
+  system.GetExpansionInterface().UpdateInterrupts();
+}
+
+CEXIBaseboard::CEXIBaseboard(Core::System& system) : IEXIDevice(system), m_position(0)
+{
+  std::string backup_filename(fmt::format("{}tribackup_{}.bin", File::GetUserPath(D_TRIUSER_IDX),
+                                          SConfig::GetInstance().GetGameID()));
+
+  m_backup = File::IOFile(backup_filename, "rb+");
+  if (!m_backup.IsOpen())
+  {
+    m_backup = File::IOFile(backup_filename, "wb+");
+  }
+
+  // Some games share the same ID Client/Server
+  if (!m_backup.IsGood())
+  {
+    PanicAlertFmt("Failed to open {}\nFile might be in use.", backup_filename.c_str());
+
+    std::srand(static_cast<u32>(std::time(nullptr)));
+
+    std::string backup_Filename =
+        fmt::format("{}tribackup_tmp_{}{}.bin", File::GetUserPath(D_TRIUSER_IDX), rand(),
+                    SConfig::GetInstance().GetGameID());
+
+    m_backup = File::IOFile(backup_filename, "wb+");
```

Comment:
`m_backup` file is created but based on the wrong variable: `backup_filename` vs `backup_Filename`.

`backup_Filename` should be renamed `backup_tmp_filename` or similar and made `const`. Otherwise, you can drop the `std::string` and just reassign `backup_filename`.